controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -688,11 +716,40 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
 
     tlvRuid->radio_uid() = network_utils::mac_from_string(ruid);
 
-    for (int i = 0; i < radio_basic_caps->maximum_number_of_bsss_supported(); i++) {
-        if (!autoconfig_wsc_add_m2(tlvwscM1)) {
+    const auto &bss_info_confs = database.get_bss_info_configuration(al_mac);
+    uint8_t num_bsses          = 0;
+
+    for (const auto &bss_info_conf : bss_info_confs) {
+        // Check if the radio supports it
+        if (!has_matching_operating_class(*radio_basic_caps, bss_info_conf)) {
+            LOG(INFO) << "Skipping " << bss_info_conf.ssid << " due to operclass mismatch";
+            continue;
+        }
+        if (!(tlvwscM1->authentication_type_flags_attr().data &
+              uint16_t(bss_info_conf.authentication_type))) {
+            LOG(INFO) << "Skipping " << bss_info_conf.ssid << " due to auth mismatch";
+            continue;
+        }
+        if (!(tlvwscM1->encryption_type_flags_attr().data &
+              uint16_t(bss_info_conf.encryption_type))) {
+            LOG(INFO) << "Skipping " << bss_info_conf.ssid << " due to encr mismatch";
+            continue;
+        }
+        if (num_bsses >= radio_basic_caps->maximum_number_of_bsss_supported()) {
+            LOG(INFO) << "Configured BSSes exceeds maximum for " << al_mac << " radio " << ruid;
+            break;
+        }
+        if (!autoconfig_wsc_add_m2(tlvwscM1, &bss_info_conf)) {
             LOG(ERROR) << "Failed setting M2 attributes";
             return false;
         }
+        num_bsses++;
+    }
+
+    // If no BSSes (either because none are configured, or because they don't match), tear down.
```

Comment:
I assumed you meant not to reach here if you called `autoconfig_wsc_add_m2(tlvwscM1, &bss_info_conf)` in line 742, but we will reach here after the previous ranges loop will be finished.

Reply:
So this should be wrapped with
```
if (!num_bsses)
   ....
```

Reply:
Oops....